### PR TITLE
chore: convert zoeken.converter package from Java to Kotlin

### PR DESCRIPTION
**Pull request by edgarvonk**
Converted zoeken.converter package from Java to Kotlin for better code maintainability and testability. 

Solves PZ-4421

### CODE REVIEW

File: src/main/java/net/atos/zac/zoeken/converter/AbstractZoekObjectConverter.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package net.atos.zac.zoeken.converter
+
+import net.atos.zac.zoeken.model.ZoekObject
+import net.atos.zac.zoeken.model.index.ZoekObjectType
+
+abstract class AbstractZoekObjectConverter<ZOEKOBJECT : ZoekObject> {
```
Comment (bas-info-nl) [inline src/main/java/net/atos/zac/zoeken/converter/AbstractZoekObjectConverter.kt:6]:
```suggestion
abstract class AbstractZoekObjectConverter<out ZOEKOBJECT : ZoekObject> {
```

File: src/main/java/net/atos/zac/zoeken/converter/AbstractZoekObjectConverter.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    abstract fun supports(objectType: ZoekObjectType): Boolean
```
Comment (bas-info-nl) [inline src/main/java/net/atos/zac/zoeken/converter/AbstractZoekObjectConverter.kt:7]: Maybe add a constructor that takes in the `ZoekObjectType` that is supported, and implement this method right here.

**Review by bas-info-nl — APPROVED** [review]
Some minor suggestions.

File: src/main/java/net/atos/zac/zoeken/converter/AbstractZoekObjectConverter.kt (hunk 1 above)
Comment (edgarvonk) [reply src/main/java/net/atos/zac/zoeken/converter/AbstractZoekObjectConverter.kt]: nice! fixed this and removed 'out' keyword from code calling these converters.

File: src/main/java/net/atos/zac/zoeken/converter/AbstractZoekObjectConverter.kt (hunk 2 above)
Comment (edgarvonk) [reply src/main/java/net/atos/zac/zoeken/converter/AbstractZoekObjectConverter.kt:7]: I might look at this later if you don't mind. :-)

**Pull request merged**